lib/internal/process/promises.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
+const { ERR_INVALID_ENV_VALUE } = require('internal/errors').codes;
+
 const { safeToString } = process.binding('util');
 
 const maybeUnhandledPromises = new WeakMap();
-const pendingUnhandledRejections = [];
-const asyncHandledRejections = [];
+let pendingUnhandledRejections = [];
+let asyncHandledRejections = [];
+const kFromPromise = true;
 let lastPromiseId = 0;
 
+const kSilent = 0;
+const kWarn = 1;
+const kErrorOnGC = 2;
+const kError = 3;
+
 exports.setup = setupPromises;
 
+function unhandledState() {
+  let env = process.env.NODE_UNHANDLED_REJECTION;
+  if (env) {
+    env = env.toUpperCase();
+    if (env === 'ERROR') {
```

Comment:
Nit: Would using a Map or a keyed collection better here? That would avoid the multiple ifs.

Reply:
I think the multiple ifs are fine since it's very unlikely to ever change

Reply:
The alternative with an object would look like:

```js
if (env) {
  env = env.toUpperCase();
  if (states[env] !== undefined) {
    return states[env];
  }
  throw new Error(...);
}
return states.WARN;
``` 

So yes, it is a bit shorter but the access for `ERROR_ON_GC` would be somewhat against our variable names. I have no strong opinion about either one.